pkg/api/validation/validation.go

```diff
@@ -444,6 +444,10 @@ func validateVolumeSource(source *api.VolumeSource, fldPath *field.Path) field.E
 			numVolumes++
 			allErrs = append(allErrs, validateISCSIVolumeSource(source.ISCSI, fldPath.Child("iscsi"))...)
 		}
```

Comment:
Validation for both "IQN"  and "InitiatorName" in the ISCSIVolumeSource were added.
As for iSCSI, we accept three type of prefixes, "iqn", "eui" and "naa".
They are supported by tagetcli. 

See man targetcli.
```
iSCSI supports multiple WWN formats: iqn, naa, and eui. Other fabrics support single formats only.
```

These are reference of the implementation.
- iqn/eui
https://www.ietf.org/rfc/rfc3721.txt

```
iqn
iqn.2000-02.edu.example.cs:users.oaks:proto.target4
iqn.1995-11.com.example.ssp:customers.4567.disks.107

Type  EUI-64 identifier
eui.02004567A425678D
```

- naa
https://tools.ietf.org/html/rfc3980#ref-FC-FS

```
Type  64bit NAA identifier (ASCII-encoded hexadecimal)
naa.52004567BA64678D

Type  128 bit NAA identifier (ASCII-encoded hexadecimal)
naa.62004567BA64678D0123456789ABCDEF
```

- python-rtslib-2.1.fb57-5.el7.noarch
```
def normalize_wwn(wwn_types, wwn):
    '''
    Take a WWN as given by the user and convert it to a standard text
    representation.

    Returns (normalized_wwn, wwn_type), or exception if invalid wwn.
    '''
    wwn_test = {
    'free': lambda wwn: True,
    'iqn': lambda wwn: \
        re.match("iqn\.[0-9]{4}-[0-1][0-9]\..*\..*", wwn) \
        and not re.search(' ', wwn) \
        and not re.search('_', wwn),
    'naa': lambda wwn: re.match("naa\.[125][0-9a-fA-F]{15}$", wwn),
    'eui': lambda wwn: re.match("eui\.[0-9a-f]{16}$", wwn),
    'ib': lambda wwn: re.match("ib\.[0-9a-f]{32}$", wwn),
    'unit_serial': lambda wwn: \
        re.match("[0-9A-Fa-f]{8}(-[0-9A-Fa-f]{4}){3}-[0-9A-Fa-f]{12}$", wwn),
    }
```